common/main/java/com/couchbase/lite/Collection.java

```diff
@@ -154,14 +154,20 @@ long getCount() throws CouchbaseLiteException {
     /**
      * Gets an existing Document object with the given ID. If the document with the given ID doesn't
      * exist in the collection, the value returned will be null.
+     * <p>
+     *
      */
     @Nullable
     public Document getDocument(@NonNull String id) throws CouchbaseLiteException {
         Preconditions.assertNotEmpty(id, "id");
         return withLockAndOpenDb(() -> {
             try { return Document.getDocument(this, id, false); }
-            catch (CouchbaseLiteException e) { Log.i(LogDomain.DATABASE, "Failed retrieving document: %s", id); }
-            return null;
+            catch (CouchbaseLiteException e) {
+                if (e.getDomain().equals(CBLError.Domain.CBLITE) && e.getCode() == CBLError.Code.NOT_FOUND) {
```

Comment:
not requesting a change, here, but FYI:
CBLError.Domain.CBLITE will never be null.  That means that saying CBLError.Domain.CBLITE.equals(...) will never NPE.